glib/tests/fileutils.c

```diff
@@ -860,10 +860,19 @@ test_stdio_wrappers (void)
 
   cwd = g_get_current_dir ();
   path = g_build_filename (cwd, "mkdir-test", NULL);
-  g_free (cwd);
   ret = g_chdir (path);
-  g_assert (errno == EACCES);
-  g_assert (ret == -1);
+  if (geteuid () == 0)
```

Comment:
Are you sure? That fix was committed 6 years ago, if it fixed this problem we wouldn't have this distro patch in the first place.